src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,24 +1,98 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+
 /**
  * <p>Реалізувати свою HashMap, а саме методи `put(K key, V value)`, `getValue()` та `getSize()`.
  * Дотриматися основних вимог щодо реалізації мапи (initial capacity, load factor, resize...)
  * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    static final int DEFAULT_CAPACITY = 16;
+    private int size;
+    private Node[] buckets;
+
+    public MyHashMap() {
+        buckets = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resize();
+        final int index = determineIndex(key);
+        if (buckets[index] == null) {
+            buckets[index] = new Node(key, value);
+            size++;
+            return;
+        }
+        Node<K, V> node = buckets[index];
+        Node<K, V> lastNode;
```

Comment:
lastNode = node
node = node.next
lastNode.next = ....
it looks like lastNode variable is redundant here

Reply:
Okay, I fixed my code here but I don`t know I do it correctly or I created a new redundant variable:)